File: Метро JS/js/Graf.js
class Graf {
  #subway;

  // входные данные
  #inputData1 = {
    lines: [
      {
        line: { color: 'red' },
        stations: [
          { x: 40, y: 490 }, // name: 'name'
          { x: 40, y: 420 },
          { x: 70, y: 350 },
          { x: 180, y: 170 },
          { x: 340, y: 280 },
          { x: 400, y: 350 },
          { x: 400, y: 500 },
          { x: 450, y: 550 },
        ],
      },
      {
        line: { color: 'blue' },
        stations: [
          { x: 20, y: 340 },
          { x: 70, y: 330 },
          { x: 200, y: 290 },
          { x: 330, y: 300 },
          { x: 500, y: 250 },
          { x: 550, y: 200 },
          { x: 570, y: 100 },
        ],
      },
      {
        line: { color: 'orange' },
        stations: [
          { x: 20, y: 100 },
          { x: 70, y: 100 },
          { x: 180, y: 130 },
          { x: 350, y: 150 },
          { x: 480, y: 230 },
          { x: 430, y: 350 },
          { x: 300, y: 500 },
        ],
      },
      {
        line: { color: 'green' },
        stations: [
          { x: 100, y: 580 },
          { x: 200, y: 550 },
          { x: 200, y: 500 },
          { x: 160, y: 400 },
          { x: 200, y: 310 },
          { x: 250, y: 250 },
          { x: 230, y: 120 },
          { x: 270, y: 80 },
          { x: 350, y: 50 },
          { x: 500, y: 150 },
        ],
      },
      {
        line: { color: 'gray' },
        stations: [
          { x: 50, y: 150 },
          { x: 220, y: 180 },
          { x: 310, y: 280 },
          { x: 400, y: 570 },
        ],
      },
      {
        line: { color: '#ad00b3' },
        stations: [
          { x: 500, y: 50 },
          { x: 400, y: 50 },
          { x: 320, y: 110 },
          { x: 250, y: 170 },
          { x: 370, y: 250 },
          { x: 500, y: 400 },
        ],
      },
      {
        line: { color: '#00969b' },
        stations: [
          { x: 600, y: 550 },
          { x: 500, y: 500 },
          { x: 410, y: 480 },
          { x: 300, y: 450 },
          { x: 200, y: 480 },
          { x: 100, y: 480 },
          { x: 80, y: 500 },
        ],
      },
      {
        line: { color: '#a8124b' },
        stations: [
          { x: 670, y: 200 },
          { x: 620, y: 300 },
          { x: 650, y: 500 },
          { x: 620, y: 550 },
          { x: 550, y: 650 },
        ],
      },
    ],
    links: [
      {
        st1: 'l0_s2',
        st2: 'l1_s1',
      },
      {
        st1: 'l0_s3',
        st2: 'l2_s2',
      },
      {
        st1: 'l0_s3',
        st2: 'l3_s6',
      },
      {
        st1: 'l0_s3',
        st2: 'l4_s1',
      },
      {
        st1: 'l0_s4',
        st2: 'l1_s3',
      },
      {
        st1: 'l0_s4',
        st2: 'l4_s2',
      },
      {
        st1: 'l0_s5',
        st2: 'l2_s5',
      },
      {
        st1: 'l1_s2',
        st2: 'l3_s4',
      },
      {
        st1: 'l1_s3',
        st2: 'l4_s2',
      },
      {
        st1: 'l1_s4',
        st2: 'l2_s4',
      },
      {
        st1: 'l2_s2',
        st2: 'l3_s6',
      },
      {
        st1: 'l4_s1',
        st2: 'l3_s6',
      },
      {
        st1: 'l5_s3',
        st2: 'l3_s6',
      },
      {
        st1: 'l5_s3',
        st2: 'l4_s1',
      },
      {
        st1: 'l6_s4',
        st2: 'l3_s2',
      },
      {
        st1: 'l6_s2',
        st2: 'l0_s6',
      },
      {
        st1: 'l7_s3',
        st2: 'l6_s0',
      },
    ],
  };

  #inputData2 = {
    lines: [
      {
        line: { color: 'red' },
        stations: [
          { x: 100, y: 550 },
          { x: 100, y: 500 },
          { x: 90, y: 400 },
          { x: 90, y: 300 },
          { x: 110, y: 250 },
          { x: 150, y: 200 },
          { x: 190, y: 180 },
          { x: 220, y: 200 },
          { x: 250, y: 250 },
          { x: 260, y: 300 },
          { x: 260, y: 350 },
          { x: 260, y: 400 },
        ],
      },
      {
        line: { color: 'blue' },
        stations: [
          { x: 40, y: 215 },
          { x: 50, y: 320 },
          { x: 105, y: 320 },
          { x: 190, y: 320 },
          { x: 275, y: 320 },
          { x: 330, y: 300 },
          { x: 390, y: 255 },
        ],
      },
      {
        line: { color: 'orange' },
        stations: [
          { x: 350, y: 105 },
          { x: 410, y: 155 },
          { x: 415, y: 245 },
          { x: 420, y: 305 },
          { x: 440, y: 385 },
        ],
      },
      {
        line: { color: 'green' },
        stations: [
          { x: 50, y: 510 },
          { x: 120, y: 520 },
          { x: 200, y: 550 },
          { x: 240, y: 570 },
        ],
      },
      {
        line: { color: 'gray' },
        stations: [
          { x: 50, y: 450 },
          { x: 30, y: 500 },
          { x: 50, y: 600 },
        ],
      },
      {
        line: { color: 'orange' },
        stations: [
          { x: 120, y: 410 },
          { x: 150, y: 450 },
          { x: 200, y: 450 },
        ],
      },
    ],
    links: [
      {
        st1: 'l0_s1',
        st2: 'l3_s1',
      },
      {
        st1: 'l0_s3',
        st2: 'l1_s2',
      },
      {
        st1: 'l0_s9',
        st2: 'l1_s4',
      },
      {
        st1: 'l1_s6',
        st2: 'l2_s2',
      },
      {
        st1: 'l3_s0',
        st2: 'l4_s1',
      },
      {
        st1: 'l0_s2',
        st2: 'l5_s0',
      },
    ],
  };

  #inputData3 = {
    lines: [
      {
        line: { color: 'red' },
        stations: [
          { x: 350, y: 600 },
          { x: 350, y: 540 },
          { x: 350, y: 400 },
          { x: 350, y: 350 },
        ],
      },
      {
        line: { color: 'blue' },
        stations: [
          { x: 330, y: 400 },
          { x: 310, y: 370 },
          { x: 300, y: 350 },
          { x: 300, y: 300 },
          { x: 330, y: 250 },
        ],
      },
      {
        line: { color: 'green' },
        stations: [
          { x: 370, y: 400 },
          { x: 490, y: 300 },
          { x: 370, y: 250 },
        ],
      },
      {
        line: { color: 'yellow' },
        stations: [
          { x: 350, y: 250 },
          { x: 350, y: 200 },
          { x: 350, y: 100 },
        ],
      },
      {
        line: { color: 'gray' },
        stations: [
          { x: 150, y: 170 },
          { x: 200, y: 120 },
          { x: 350, y: 70 },
          { x: 400, y: 60 },
          { x: 450, y: 50 },
        ],
      },
    ],
    links: [
      {
        st1: 'l0_s2',
        st2: 'l1_s0',
      },
      {
        st1: 'l0_s2',
        st2: 'l2_s0',
      },
      {
        st1: 'l2_s2',
        st2: 'l3_s0',
      },
      {
        st1: 'l1_s4',
        st2: 'l3_s0',
      },
      {
        st1: 'l3_s2',
        st2: 'l4_s2',
      },
    ],
  };

  constructor() {
    let canvas = document.getElementById('metro-map-canvas');
    canvas.width = canvas.clientWidth;
    canvas.height = canvas.clientHeight;
    let ctx = canvas.getContext('2d');

    let inputData;
    let map = window.location.search.replace(/.+map=(.+)&id=(.+)/, '$1');
    if (map == '') map = 'map1';
    document.getElementById('map').value = map;

    if (map == 'map1') inputData = this.#inputData1;
    if (map == 'map2') inputData = this.#inputData2;
    if (map == 'map3') inputData = this.#inputData3;

    this.#subway = new Subway(ctx, inputData.lines, inputData.links);
  }
}

const graf = new Graf();

File: Metro C#/Metro/Class/Graf.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Media;

namespace Metro
{
    public static class Graf
    {
        public static readonly ((SolidColorBrush color, (int x, int y)[] stations)[] lines, (string st1, string st2)[] links)[] inputData = new[]{
            (new[]{
                (Brushes.Red,  new[]{
                (350,600),
                (350,540),
                (350,400),
                (350,350)
            }),
                (Brushes.Blue,  new[]{
                (330, 420),
                (310, 370),
                (300, 350),
                (300, 300),
                (330, 280),
            }),
                (Brushes.Green,  new[]{
                (370, 420),
                (490, 300),
                (370, 280),
            }),
                (Brushes.Yellow,  new[]{
                (350, 250),
                (350, 200),
                (350, 100),
            }),
                (Brushes.Gray,  new[]{
                    (150, 170),
                    (200, 120),
                    (350, 70),
                    (400, 60),
                    (450, 50),
                })
            }, new[]{
                ("l0_s2", "l1_s0"),
                ("l0_s2", "l2_s0"),
                ("l2_s2", "l3_s0"),
                ("l1_s4", "l3_s0"),
                ("l3_s2", "l4_s2"),
                ("l1_s4", "l2_s2")
            }),
            (new[]{
                (Brushes.Red, new[]{
                    (100, 550),
                    (100, 500),
                    (90, 400),
                    (90, 300),
                    (110, 250),
                    (150, 200),
                    (190, 180),
                    (220, 200),
                    (250, 250),
                    (260, 300),
                    (260, 350),
                    (260, 400)
                }),
                (Brushes.Blue, new[]{
                    (40, 215),
                    (50, 320),
                    (105, 320),
                    (190, 320),
                    (275, 320),
                    (330, 300),
                    (390, 255),
                }),
                (Brushes.Orange, new[]{
                    (350, 105),
                    (410, 155),
                    (415, 245),
                    (420, 305),
                    (440, 385),
                }),
                (Brushes.Green, new[]{
                    (50, 510),
                    (120, 520),
                    (200, 550),
                    (240, 570),
                }),
                (Brushes.Gray, new[]{
                    (50, 450),
                    (30, 500),
                    (50, 600),
                }),
                (Brushes.Coral, new[]{
                    (120, 410),
                    (150, 450),
                    (200, 450),
                })
            }, new[]{
                ("l0_s1", "l3_s1"),
                ("l0_s3", "l1_s2"),
                ("l0_s9", "l1_s4"),
                ("l1_s6", "l2_s2"),
                ("l3_s0", "l4_s1"),
                ("l0_s2", "l5_s0"),
            }),
            (new[]{
                (Brushes.Red, new[]{
                    (40, 490),
                      (40, 420),
                      (70, 350),
                      (180, 170),
                      (340, 280),
                      (400, 350),
                      (400, 500),
                      (450, 550),
                }),
                (Brushes.Blue, new[]{
                    (20, 330),
                      (70, 320),
                      (190, 280),
                      (330, 300),
                      (500, 250),
                      (550, 200),
                      (570, 100),
                }),
                (Brushes.Orange, new[]{
                    (20, 100),
                      (70, 100),
                      (180, 130),
                      (350, 150),
                      (480, 230),
                      (430, 350),
                      (300, 500),
                }),
                (Brushes.Green, new[]{
                    (100, 580),
                      (200, 550),
                      (200, 500),
                      (160, 400),
                      (200, 310),
                      (250, 250),
                      (230, 120),
                      (270, 80),
                      (350, 50),
                      (500, 150),
                }),
                (Brushes.Gray, new[]{
                    (50, 150),
                      (220, 210),
                      (310, 280),
                      (400, 570),
                }),
                (Brushes.Brown, new[]{
                    (500, 50),
                      (400, 50),
                      (320, 110),
                      (250, 170),
                      (370, 250),
                      (500, 400),
                }),
                (Brushes.DarkCyan, new[]{
                    (590, 550),
                      (500, 510),
                      (410, 470),
                      (300, 460),
                      (200, 470),
                      (100, 480),
                      (70, 500),
                }),
                (Brushes.DarkRed, new[]{
                    (670, 200),
                      (620, 300),
                      (650, 500),
                      (620, 550),
                      (550, 650),
                })
            }, new[]{
                ("l0_s2", "l1_s1"),
                ("l0_s3", "l2_s2"),
                ("l0_s3", "l3_s6"),
                ("l0_s3", "l4_s1"),
                ("l0_s4", "l1_s3"),
                ("l0_s4", "l4_s2"),
                ("l0_s5", "l2_s5"),
                ("l1_s2", "l3_s4"),
                ("l1_s3", "l4_s2"),
                ("l1_s4", "l2_s4"),
                ("l2_s2", "l3_s6"),
                ("l4_s1", "l3_s6"),
                ("l5_s3", "l3_s6"),
                ("l5_s3", "l4_s1"),
                ("l6_s4", "l3_s2"),
                ("l6_s2", "l0_s6"),
                ("l7_s3", "l6_s0"),
            })
        };
    }
}